golang/pkg/events/events.go

```diff
@@ -1,24 +1,53 @@
 package events
 
 import (
+	"github.com/google/uuid"
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/anypb"
+	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
 )
 
-func ToEvent(event proto.Message, source string, id string) ([]byte, error) {
+type CloudEventOptions struct {
+	Id     string
+	Source string
+	Time   *timestamppb.Timestamp
```

Comment:
Now seeing the elixir comparison, it would be better to have this as standard golang time field. And we do the conversion. This way, the user of this functions don't need to import any protobuf package, and the conversion would happen in this code

Reply:
I agree, the user shouldn't care about the underlying serializer.

Reply:
Agreed, I'll address this together with Fabrizio's suggestion to use the functional options pattern